Python/chapter01/1.9 - String Rotation/miguel_1.9_sol.py

```diff
@@ -25,15 +25,9 @@ def string_rotation(s1: str, s2: str) -> bool:
     """
     Given two strings, string_rotation will check if s2 is a rotation of s1
     using only one call to isSubstring.
-    We will loop through the original string and compare chars of
-    rotated string.  At the first character equality, we will
-    compare the rest of the original string to the rotated
-    string's substring of the same length.
-    If they match, then we save the index of the
-    rotated string + 1, and if whatever is left of
-    s1 is a substring of s2, then we must have a
-    rotation since we compared the complementary part already.
-    Runtime: Worst case:  O(n^2)
+    If s2 is a rotation of s1, then when we add s1 to itself, s2
+    must be a substring of s1 + s1
+    Runtime: O(n)
```

Comment:
can you back this claim?

Reply:
sorry, should have been more clear: i don't think O(n) is accurate. can you explain how you arrived at that?